src/lib/components/header/QHeader.svelte

```diff
@@ -1,53 +1,63 @@
-<svelte:options runes={true} />
-
 <script lang="ts">
-  import { untrack } from "svelte";
-  import { createClasses } from "$lib/utils";
-  import { QToolbar } from "$lib";
+  import { getContext, onDestroy, untrack } from "svelte";
+  import QToolbar from "$components/toolbar/QToolbar.svelte";
   import QContext from "$lib/classes/QContext.svelte";
-  import type { LayoutContext } from "../layout/QLayout.svelte";
+  import QScrollObserver from "$lib/classes/QScrollObserver.svelte";
+  import type { AppbarContext } from "$components/layout/QLayout.svelte";
+  import type { QLayoutProps } from "$components/layout/props";
   import type { QHeaderProps } from "./props";
 
+  const ID = Date.now();
+
   let {
+    elevated = false,
     inset = false,
-    elevate = false,
-    border = false,
+    reveal = false,
+    revealOffset = 250,
+    height = 64,
     children,
     ...props
   }: QHeaderProps = $props();
 
-  const ctx = QContext.get<LayoutContext>("layout");
+  const headerContext = QContext.get<AppbarContext>("QHeader");
+  const layoutView = getContext<{ value: NonNullable<QLayoutProps["view"]> }>("view");
+  if (!headerContext || !layoutView) {
+    throw new Error("QHeader should be used inside QLayout");
+  }
+
+  const scroll = $derived(
+    reveal ? new QScrollObserver(`.q-header--${ID} ~ .q-layout__content`) : undefined
+  );
+  const offset = $derived(scroll ? scroll.position - height : undefined);
+  // Collapse the header `${reavealOffset}px` below the top of layout content when scrolling down
+  const collapsed = $derived(reveal && scroll?.direction === "down" && offset! - revealOffset > 0);
```

Comment:
Here, it's needed because `scroll` exists only if reveal exists, then `offset` exists too. Here, we check if `reveal` exists so offset exists for sure